model/basic/basic_model.py

```diff
@@ -59,26 +56,61 @@ def __init__(self, config, session):
         self.training_epochs = config['training_epochs']
         self.users_to_select = config['users_to_select']
 
+        if 'use_l2_loss' in config:
```

Comment:
With the new template system these checks can be removed as all models will have configurations for everything, even things that won't be used.